dio/lib/src/options.dart

```diff
@@ -698,24 +714,25 @@ class _RequestConfig {
 
   Duration? _receiveTimeout;
 
-  /// The request Content-Type. The default value is [ContentType.json].
+  /// The request Content-Type. Defaults to [ContentType.json] if capable.
+  ///
   /// If you want to encode request body with 'application/x-www-form-urlencoded',
-  /// you can set `ContentType.parse('application/x-www-form-urlencoded')`, and [Dio]
-  /// will automatically encode the request body.
+  /// you can set `ContentType.parse('application/x-www-form-urlencoded')`,
+  /// and [Dio] will automatically encode the request body.
+  String? get contentType => _headers[Headers.contentTypeHeader] as String?;
+
   set contentType(String? contentType) {
-    if (contentType != null) {
-      _headers[Headers.contentTypeHeader] =
-          _defaultContentType = contentType.trim();
+    final newContentType = contentType?.trim();
+    _defaultContentType = newContentType;
```

Comment:
Whether it is null or not, the value of _defaultContentType will be updated, is this reasonable?

Reply:
It's equivalent to the previous code.
```dart
_headers[Headers.contentTypeHeader] = _defaultContentType = contentType.trim();
```
This will assign the content type to both `_headers[Headers.contentTypeHeader]` and `_defaultContentType`.